svsim/src/main/scala/Workspace.scala

```diff
@@ -36,7 +36,24 @@ object Workspace {
       FileVisitResult.CONTINUE
     }
   }
+
+  /**
+  * Detect the project root (workspace) directory. It uses the "chisel.project.root" Java system property if it is set.
+  *   Otherwise, checks the environment variables CHISEL_PROJECT_ROOT (can be set by user) and MILL_WORKSPACE_ROOT (set by mill).
+  * If none of those work, returns the current working directory.
+  *
+  * @return the absolute path to the project root or the current working directory
+  */
+  def getProjectRootOrCwd: Path =
+    sys.props
+      .get("chisel.project.root")
+      .orElse(sys.env.get("CHISEL_PROJECT_ROOT"))
```

Comment:
Should we consider enshrining `CHISEL_PROJECT_ROOT` elsewhere like here: https://github.com/chipsalliance/chisel/blob/ff99b21a2f88bef85f07f4a6bb373ed4c174de5b/core/src/main/scala-2/chisel3/internal/SourceInfoFileResolver.scala#L18

Reply:
I think it's good to have consistent behavior, but I'd do that in a separate PR.
For the record, the reason for having such an environment variable is that the "native" versions of mill (and bloop), at least on macOS, seem to ignore `-Dprop=value` both from `.mill-jvm-opts` and the command line. Therefore, it would be helpful to provide an alternative method for specifying this.

Reply:
That's annoying that native is ignoring `-Dprop`, will have to look into that at some point, would you mind filing an issue about that so when I forget I eventually notice the issue and remember again?